packages/block-editor/src/components/list-view/block-select-button.js

```diff
@@ -77,6 +83,14 @@ function ListViewBlockSelectButton(
 		  )
 		: '';
 
+	const [ isRenamingBlock, setBlockBeingRenamed ] = useState( false );
```

Comment:
Oh that's counterintuative but useful knowledge. Thank you.

I would have expected that moving state up would result in entire tree's being re-rendered for changes to a single node.

I'm happy to do this _if_ we get a consensus on this as a UX.

Reply:
It's definitely not a blocker and I've found the performance issues with the list view can be quite subtle and wind up being a trade-off of where we want to place the performance hit.

> I would have expected that moving state up would result in entire tree's being re-rendered for changes to a single node.

Yes — it's a tricky one, because local state is typically better for when we're making a change to one row and we don't want to re-render other rows, as you mention, whereas state at the root is helpful for when we want to reduce what's happening at the individual row level. To take the idea one step further (if focusing purely on performance), we could potentially move the `ListViewBlockRenameUI` up to also be rendered at the root of the list view, but have its Popover be positioned to the selected block — in that case, we'd need to go looking for the desired element to anchor the popover to when it is opened.

For now, especially since some of these performance optimisations can result in harder to read code, I think it's probably fine to just leave things with local to the block-select-button state. I mostly wanted to add a couple of comments so that if we do notice any performance issues further down the track, we've got a bit of a discussion going here so we know where to look 🙂

Separately to all this, I have an open issue (https://github.com/WordPress/gutenberg/issues/55114) for improving the fluidity of the list view windowing logic that I'd like to look into at some point — the goal there being to more gracefully show/hide list view items so that if there is a performance issue on mount, it's not as apparent to a user.

For now, though, feel free to close out this particular discussion thread!